backend/core/views.py

```
@@ -3417,44 +3417,88 @@ def mailing(self, request, pk):
-                    ]
+
+                # Collect all source requirement assessment IDs
+                source_assessment_ids = [
+                    assessment.mapping_inference["source_requirement_assessment"]["id"]
+                    for assessment in computed_assessments
+                ]
+
+                # Fetch all baseline requirement assessments in one query
+                baseline_assessments = {
+                    str(ra.id): ra
+                    for ra in RequirementAssessment.objects.filter(
```

Ruff (0.8.2)

`RequirementAssessment` may be undefined, or defined from star imports

(F405)